### PR TITLE
[Color system] Apply color system to TextStyle

### PR DESCRIPTION
**Pull request by dleroux**
addresses: https://github.com/Shopify/polaris-ux/issues/366

Screen Shot 2019-11-08 at 8 55 37 AM
Screen Shot 2019-11-08 at 8 55 22 AM

### WHAT is this pull request doing?

- Not sure if this is the right approach but I updated the shadowFromDirectLight since it's should look like a border in dark mode.

- Not the success green. It isn't very visible in light mode and really bright in dark mode. Is that correct?
 
## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page, TextStyle, Stack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Stack>
        <TextStyle variation="negative">Negative</TextStyle>
        <TextStyle variation="positive">Positive</TextStyle>
        <TextStyle variation="strong">Strong</TextStyle>
        <TextStyle>Regular</TextStyle>
        <TextStyle variation="subdued">Subdued</TextStyle>
        <TextStyle variation="code">Code</TextStyle>
      </Stack>
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the [Polaris UI kit](https://polaris.shopify.com/resources/polaris-ui-kit)

### CODE REVIEW

File: src/components/TextStyle/TextStyle.scss [hunk 1]
```diff
@@ -4,21 +4,21 @@ $code-padding: 0 spacing(extra-tight);
...
-  color: color('red', 'dark');
+  color: var(--p-critical-text, color('red', 'dark'));
 }
 
 .variationCode {
   position: relative;
   padding: $code-padding;
   border-radius: border-radius();
-  background-color: color('sky', 'light');
+  background-color: var(--p-surface-background, color('sky', 'light'));
```
Comment (dleroux) [inline src/components/TextStyle/TextStyle.scss]: Would this need to become `--p-surface-background` or `--p-surface-foreground` depending on where it's used?

Comment (tmlayton) [reply src/components/TextStyle/TextStyle.scss]: I think it should be `--p-surface-foreground-subdued`

**Comment by tmlayton** [comment]
> fixes: Shopify/polaris-ux#366

If you don’t mind changing that to `addresses` as this is only half of the work, still need to do Rails too.

File: src/components/TextStyle/TextStyle.scss [hunk 2]
```diff
@@ -4,21 +4,21 @@ $code-padding: 0 spacing(extra-tight);
...
   display: inline-block;
   font-size: $code-font-size;
-  box-shadow: inset 0 0 0 1px color('sky');
+  box-shadow: inset 0 0 0 1px var(--p-shadow-from-direct-light, color('sky'));
```
Comment (tmlayton) [inline src/components/TextStyle/TextStyle.scss]: I think since this is supposed to emulate a light border it should be `--p-border-subdued-on-surface`

**Comment by tmlayton** [comment]
> Not the success green. It isn't very visible in light mode and really bright in dark mode. Is that correct?

I think for both success and critical, the values should be:

```ts
light: {lightness: 40},
dark: {lightness: 60},
```

This will still be AA even on the most muted surfaces (lightness of 90/10 respectively for light/dark). AAA = a lightness difference of ~62 or more and AA = a lightness difference of ~50 or more, regardless the hue or saturation.

/cc @jessebc 

Screen Shot 2019-11-08 at 10 06 36 PMScreen Shot 2019-11-08 at 10 06 42 PM

**Comment by jessebc** [comment]
Will these text colours propagate through all of the components now?

**Comment by dleroux** [comment]
> Will these text colours propagate through all of the components now?

Yes, I believe they would.

**Pull request merged**